librz/core/analysis_tp.c

```diff
@@ -917,9 +917,11 @@ RZ_API void rz_core_analysis_type_match(RzCore *core, RzAnalysisFunction *fcn, H
 			rz_list_free(fcns);
 		}
 	}
+
 	// Type propagation for register based args
+	RzPVector *cloned_vars = (RzPVector *)rz_vector_clone((RzVector *)&fcn->vars);
 	void **vit;
-	rz_pvector_foreach (&fcn->vars, vit) {
+	rz_pvector_foreach (cloned_vars, vit) {
```

Comment:
Wouldn't this uaf if a var is removed inside the loop? The variable would not only be removed from `fcn->vars`, but also freed.

Reply:
In theory it will.

And `resolve_var_overlaps` using the same method seems to be no problem, my fault.

https://github.com/rizinorg/rizin/blob/21584e416cdcef2fa7d855c5aabf592a965f0e8d/librz/analysis/var.c#L96-L108